nilearn/maskers/nifti_masker.py

```diff
@@ -100,6 +100,8 @@ def _filter_and_mask(
     # as small as possible in order to speed up the process
 
     if not _utils.niimg_conversions._check_same_fov(imgs, mask_img_):
+        warnings.warn('imgs are being upsampled to the mask_img resolution, '
+            'you might want to provide a target_affine to save memory and computation time.')
```

Comment:
```suggestion
        warnings.warn(
            'imgs are being upsampled to the mask_img resolution, '
            'you might want to provide a target_affine to save memory and computation time.',
            UserWarning
        )
```

We can also probably improve this warning a bit. Maybe adding that this is memory intensive when using a high resolution mask.

Reply:
@ymzayek  How about "imgs are being upsampled to the mask_img resolution. This process is memory intensive when using a high resolution mask. You might want to provide a target_affine to save memory and computation time." ?